src/HttpAsyncClient.php

```diff
@@ -23,5 +23,5 @@
      *
      * @throws \Exception If processing the request is impossible (eg. bad configuration).
      */
-    public function sendAsyncRequest(RequestInterface $request);
+    public function sendAsyncRequest(RequestInterface $request): Promise;
```

Comment:
Is this problem when we are doing the same update for the Async HTTP client PSR?

Reply:
there needs to be a new major version anyways when a promise PSR is out, so we might as well do this change now to make it more explicit.

Reply:
I would like to be more permissive in the async implementation, to allow every value and exception in our promise, this would be better, after years of use it was a mistake to reduce the scope IMO.

Reply:
Fair, so what do you suggest? I've written `Promise` here, not `HttpPromise`.

Reply:
In fact, it's in another package, we would need to update the promise lib and then allow 2.0 here, but maybe it will not be bc break for this package, so nothing to change here

Reply:
(We would still need to release a 2.1 with the package dependency upgrade to allow 1.0 || 2.0 for the promise lib)

Reply:
In fact what do you think of using https://amphp.org/amp/promises/ interface ? there is an adapter for react and we could drop support of async for guzzle (as it's not really async in fact, it's only parallel)

Reply:
Hm.. Im not sure at all...

On one hand: Do as little change as possible. That will make it easier for everyone. 
On the other hand: Can we make the "correct" decision now so there is no need for v3 when we got a PSR for Promise...? Probably not, right?